packages/datatrak-web/src/api/mutations/useSubmitSurveyResponse.ts

```diff
@@ -52,7 +58,7 @@ export const useSubmitSurveyResponse = (fromLocation?: string | undefined) => {
       }
 
       return post('submitSurveyResponse', {
-        data: { ...surveyResponseData, answers },
+        data: { ...surveyResponseData, answers: { ...answers, ...entityQuestionAutoFill } },
```

Comment:
I like this idea of adding the response at the end. Just double checking: are there any occasions where the primary entity determines the visibility of any other questions? Technically it can I happen but I don't know that it ever does

Reply:
I couldn't find any examples but I decided to move the autofill behaviour to the form context anyway so that it now happens at the start of the survey